modules/github_integration/lib/api/v3/github_integration/github_pull_request_representer.rb

```diff
@@ -0,0 +1,166 @@
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+require 'roar/decorator'
+require 'roar/json/hal'
+
+module API
+  module V3
+    module GithubIntegration
+      class GithubPullRequestRepresenter < ::API::Decorators::Single
+        include API::Caching::CachedRepresenter
+        include API::Decorators::DateProperty
+        include API::Decorators::LinkedResource
+
+        cached_representer key_parts: %i[github_user merged_by latest_check_runs],
+                           disabled: false
+
+        link :staticPath do
+          next
+        end
+
+        property :id
+
+        property :number,
+                 setter: ->(*) { nil }
+
+        property :github_html_url,
+                 setter: ->(*) { nil }
+
+        property :state,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :repository,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        date_time_property :github_updated_at,
+                           render_nil: true,
+                           setter: ->(*) { nil }
+
+        property :title,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :body,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :draft,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :merged,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :merged_at,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :comments_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :review_comments_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :additions_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :deletions_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :changed_files_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :labels,
+                 setter: ->(*) { nil }
+
+        property :github_user,
+                 getter: ->(*) {
+                   {
+                     login: github_user.github_login,
+                     htmlUrl: github_user.github_html_url,
+                     avatarUrl: github_user.github_avatar_url
+                   }
+                 },
+                 setter: ->(*) { nil }
+
+        property :merged_by,
+                 getter: ->(*) {
+                   next unless merged_by
+
+                   {
+                     login: merged_by.github_login,
+                     htmlUrl: merged_by.github_html_url,
+                     avatarUrl: merged_by.github_avatar_url
+                   }
+                 },
```

Comment:
Same as above.

Reply:
not exactly sure how to best approach that (although I agree that would be better), see https://github.com/opf/openproject/pull/9070#discussion_r611972158